river/datasets/synth/friedman.py

```diff
@@ -51,6 +52,230 @@ def __iter__(self):
         while True:
 
             x = {i: rng.uniform(a=0, b=1) for i in range(10)}
-            y = 10 * math.sin(math.pi * x[0] * x[1]) + 20 * (x[2] - .5) ** 2 + 10 * x[3] + 5 * x[4]
+            y = 10 * math.sin(math.pi * x[0] * x[1]) + 20 * (x[2] - .5) ** 2 + 10 * x[3] \
+                + 5 * x[4] + rng.gauss(mu=0, sigma=1)
 
             yield x, y
+
+
+class FriedmanDrift(base.SyntheticDataset):
+    """Friedman synthetic dataset with concept drifts.
+
+    Each observation is composed of 10 features. Each feature value is sampled uniformly in [0, 1].
+    Only the first 5 features are relevant. The target is defined by different functions depending
+    on the type of the drift.
+
+    The three available modes of operation of the data generator are described in [^1].
+
+    Parameters
+    ----------
+    drift_type
+        The variant of concept drift.</br>
+        - `'lea'`: Local Expanding Abrupt drift. The concept drift appears in two distinct
+        regions of the instance space, while the remaining regions are left unaltered.
+        There are three points of abrupt change in the training dataset.
+        At every consecutive change the regions of drift are expanded.</br>
+        - `'gra'`: Global Recurring Abrupt drift. The concept drift appears over the whole
+        instance space. There are two points of concept drift. At the second point of drift
+        the old concept reoccurs.</br>
+        - `'gsg'`: Global and Slow Gradual drift. The concept drift affects all the instance
+        space. However, the change is gradual and not abrupt. After each one of the two change
+        points covered by this variant, and during a window of length `transition_window`,
+        examples from both old and the new concepts are generated with equal probability.
+        After the transition period, only the examples from the new concept are generated.
+    change_point1
```

Comment:
This can be a single parameter:
```python
position: tuple = (point_1, point_2, point_3)
```

Reply:
This simplifies the usage for the user, indeed. I changed how the concept drift locations are set in commit [afa307d](https://github.com/online-ml/river/pull/387/commits/afa307da973ab1519b2d06ae3852ba566f5ee147), as suggested.